format/format.go

```diff
@@ -1356,3 +1385,36 @@ func ensureFutureKeywordImport(imps []*ast.Import, kw string) []*ast.Import {
 	imp.Location = defaultLocation(imp)
 	return append(imps, imp)
 }
+
+// ArgErrDetail but for `fmt` checks since compiler has not run yet.
+type ArityFormatErrDetail struct {
+	Have []string `json:"have"`
+	Want []string `json:"want"`
+}
+
+// arityMismatchError but for `fmt` checks since the compiler has not run yet.
+func ArityFormatMismatchError(operands []*ast.Term, operator string, loc *ast.Location, f *types.Function) *ast.Error {
+	want := make([]string, len(f.Args()))
+	for i := range f.Args() {
+		want[i] = types.Sprint(f.Args()[i])
+	}
+
+	have := make([]string, len(operands))
+	for i := 0; i < len(operands); i++ {
+		have[i] = ast.TypeName(operands[i].Value)
+	}
+	err := ast.NewError(ast.TypeErr, loc, "%s: %s", operator, "arity mismatch")
+	err.Details = &ArityFormatErrDetail{
+		Have: have,
+		Want: want,
+	}
+	return err
+}
+
+// Lines returns the string representation of the detail.
+func (d *ArityFormatErrDetail) Lines() []string {
+	lines := make([]string, 2)
+	lines[0] = "have: " + "(" + strings.Join(d.Have, ",") + ")"
+	lines[1] = "want: " + "(" + strings.Join(d.Want, ",") + ")"
+	return lines
```

Comment:
[nit] It doesn't make much of a difference, but you could also do
```suggestion
	return []string{
		"have: " + "(" + strings.Join(d.Have, ",") + ")",
		"want: " + "(" + strings.Join(d.Want, ",") + ")",
	}
```